hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/AbstractOmBucketReadWriteOps.java

```diff
@@ -0,0 +1,241 @@
+
+  @Option(names = {"-W", "--num-of-write-operations"},
+      description = "Number of write operations to be performed by each " +
+          "thread.",
+      defaultValue = "10")
+  private int numOfWriteOperations;
+
+  private OzoneConfiguration ozoneConfiguration;
+  private Timer timer;
+  private ContentGenerator contentGenerator;
+  private int readThreadCount;
+  private int writeThreadCount;
+
+  protected abstract void display();
+
+  protected abstract void initialize(OzoneConfiguration configuration)
+      throws Exception;
+
+  @Override
+  public Void call() throws Exception {
+    init();
+
+    readThreadCount = (readThreadPercentage * totalThreadCount) / 100;
+    writeThreadCount = totalThreadCount - readThreadCount;
+
+    display();
+    print("SizeInBytes: " + sizeInBytes);
+    print("bufferSize: " + bufferSize);
+    print("totalThreadCount: " + totalThreadCount);
+    print("readThreadPercentage: " + readThreadPercentage);
+    print("writeThreadPercentage: " + (100 - readThreadPercentage));
+    print("readThreadCount: " + readThreadCount);
+    print("writeThreadCount: " + writeThreadCount);
+    print("numOfReadOperations: " + numOfReadOperations);
+    print("numOfWriteOperations: " + numOfWriteOperations);
+
+    ozoneConfiguration = createOzoneConfiguration();
+    contentGenerator = new ContentGenerator(sizeInBytes, bufferSize);
+    timer = getMetrics().timer("om-bucket-read-write-ops");
+
+    initialize(ozoneConfiguration);
+
+    return null;
+  }
+
+  protected abstract String createPath(String path) throws IOException;
+
+  protected int readOperations(int keyCountForRead) throws Exception {
+
+    // Create keyCountForRead/fileCountForRead (defaultValue = 1000) keys/files
+    // under rootPath/readPath
+    String readPath = createPath("readPath");
+    create(readPath, keyCountForRead);
+
+    // Start readThreadCount (defaultValue = 90) concurrent read threads
+    // performing numOfReadOperations (defaultValue = 50) iterations
+    // of read operations (bucket.listKeys(readPath) or
+    // fileSystem.listStatus(rootPath/readPath))
+    ExecutorService readService = Executors.newFixedThreadPool(readThreadCount);
+    CompletionService<Integer> readExecutorCompletionService =
+        new ExecutorCompletionService<>(readService);
+    List<Future<Integer>> readFutures = new ArrayList<>();
+    for (int i = 0; i < readThreadCount; i++) {
+      readFutures.add(readExecutorCompletionService.submit(() -> {
+        int readCount = 0;
+        try {
+          for (int j = 0; j < numOfReadOperations; j++) {
+            readCount = getReadCount(readCount, "readPath");
+          }
+        } catch (IOException e) {
+          LOG.warn("Exception while listing keys/files ", e);
+        }
+        return readCount;
+      }));
+    }
+
+    int readResult = 0;
+    for (int i = 0; i < readFutures.size(); i++) {
+      try {
+        readResult += readExecutorCompletionService.take().get();
```

Comment:
@tanvipenumudy we don't need this try-catch block. Lets throw exception back to the caller.

   ```
   } catch (InterruptedException e) {
        e.printStackTrace();
      } catch (ExecutionException e) {
        e.printStackTrace();
      }
```

Reply:
Thank you for the comment, I have removed the try-catch block.